lib/chef/data_collector/messages.rb

```diff
@@ -0,0 +1,125 @@
+#
+# Author:: Adam Leff (<adamleff@chef.io)
+# Author:: Ryan Cragun (<ryan@chef.io>)
+#
+# Copyright:: Copyright 2012-2016, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "json"
+require "securerandom"
+require_relative "messages/helpers"
+
+class Chef
+  class DataCollector
+    module Messages
+      extend Helpers
+
+      #
+      # Message payload that is sent to the DataCollector server at the
+      # start of a Chef run.
+      #
+      # @param run_status [Chef::RunStatus] The RunStatus instance for this node/run.
+      #
+      # @return [Hash] A hash containing the run start message data.
+      #
+      def self.run_start_message(run_status)
```

Comment:
well, for the record i tend to disagree.  we can create other objects, though, which just call the methods here.